File: invoice-api/src/main/java/com/invoiceapi/services/interfaces/InvoiceService.java
package com.invoiceapi.services.interfaces;

import com.invoiceapi.models.dtos.InvoiceDto;
import com.invoiceapi.models.entities.Client;
import com.invoiceapi.models.entities.Invoice;
import com.invoiceapi.models.response.BusinessResponse;
import com.invoiceapi.models.response.InvoiceResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface InvoiceService {
    InvoiceResponse createInvoice(InvoiceDto invoiceDto, String email);

    InvoiceResponse getInvoiceById(Long invoiceId);

    InvoiceResponse updateInvoice(Long id, InvoiceDto invoiceDto, String email);

    void deleteInvoice(Long id, String email);

    List<InvoiceResponse> getAllInvoices(String email);
    void updateInvoiceStatus(Long invoiceId, String status, String email);

    Page<InvoiceResponse> getInvoices(String email, Pageable pageable);
    Page<InvoiceResponse> getInvoicesByClient(Long clientId, String email, Pageable pageable);
    Page<InvoiceResponse> searchInvoicesByClientName(String clientName, String email, Pageable pageable);
}

File: invoice-front/src/services/invoiceService.js
import client from "@/helpers/client.js";

const invoiceService = {
    async createInvoice(invoiceData) {
        try {
            const response = await client.post('/invoices', invoiceData);
            return response.data;
        } catch (error) {
            throw error;
        }
    },
    async updateInvoice(id, invoiceData) {
        try {
            const response = await client.put(`/invoices/${id}`, invoiceData);
            return response.data;
        } catch (error) {
            throw error;
        }
    },
    async deleteInvoice(id) {
        try {
            await client.delete(`/invoices/${id}`);
        } catch (error) {
            throw error;
        }
    },
    async getInvoiceById(id) {
        try {
            const response = await client.get(`/invoices/${id}`);
            return response.data;
        } catch (error) {
            throw error;
        }
    },
    async getAllInvoices(page = 0,
                         sizePerPage = 5,
                         sortBy = 'updatedAt',
                         sortDirection = 'desc') {
        try {
            const response = await client.get('/invoices', {
                params: {page, sizePerPage, sortBy, sortDirection},
            });
            return response.data;
        } catch (error) {
            throw error;
        }
    },
    async getAllListInvoices() {
        try {
            const response = await client.get('/invoices/list');
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    async getAllClientInvoices(clientId,
                               page = 0,
                               sizePerPage = 5,
                               sortBy = 'updatedAt',
                               sortDirection = 'desc') {
        try {
            const response = await client.get(`/invoices/client/${clientId}`, {
                params: {page, sizePerPage, sortBy, sortDirection},
            });
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    async searchInvoicesByClientName(clientName,
                                     page = 0,
                                     sizePerPage = 5) {
        try {
            const response = await client.get(`/invoices/search`, {
                params: {clientName, page, sizePerPage},
            });
            return response.data;
        } catch (error) {
            throw error;
        }
    },

    async updateInvoiceStatus(invoiceId, status) {
        try {
            const response = await client.patch(`/invoices/${invoiceId}/status`, {status});
            return response.data;
        } catch (error) {
            throw error;
        }
    },
};


export default invoiceService;